app/controllers/parties_controller.rb

```diff
@@ -1,60 +1,57 @@
 class PartiesController < ApplicationController
-
   def index
-    @parties = order_list(Party.eager_all, :name)
-
-    format({ serialized_data: @parties })
+    @parties = Parliament::Request.new.parties.get
   end
 
   def current
-    @parties = order_list(Party.eager_all('current'), :name)
-
-    format({ serialized_data: @parties })
+    @parties = Parliament::Request.new.parties.current.get
   end
 
   def show
-    @party = Party.eager_find(params[:id]) or not_found
+    id = params[:id]
```

Comment:
this has been done. Please see new pull request for branch: aebirim/new-changes-party-controller